### PR TITLE
Modified pycbc_plot_singles_vs_params to make plotting neat by removi…

### PR DESCRIPTION
**Pull request by sum33it**
Allow any single detector ranking statistic to plot using pycbc_plot_singles_vs_params

### CODE REVIEW

File: bin/hdfcoinc/pycbc_plot_singles_vs_params [hunk 1]
```diff
@@ -127,16 +127,20 @@ if opts.z_var == 'density':
...
-        z = trigs.snr
-    elif opts.z_var == 'max(newsnr)':
-        z = trigs.newsnr
-    elif opts.z_var == 'max(newsnr_sgveto)':
-        z = trigs.newsnr_sgveto
-    elif opts.z_var == 'max(newsnr_sgveto_psdvar)':
-        z = trigs.newsnr_sgveto_psdvar
+    z = trigs.'%s'%opts.z_var    
+
+#elif opts.z_var in ['max(snr)', 'max(newsnr)', 'max(newsnr_sgveto)', 'max(newsnr_sgveto_psdvar)']:
```
Comment (ahnitz) [inline bin/hdfcoinc/pycbc_plot_singles_vs_params]: Hey, let's delete these comments.

File: bin/hdfcoinc/pycbc_plot_singles_vs_params [hunk 2]
```diff
@@ -127,16 +127,20 @@ if opts.z_var == 'density':
...
     cb_style = {}
-    if opts.z_var == 'max(snr)':
-        z = trigs.snr
-    elif opts.z_var == 'max(newsnr)':
-        z = trigs.newsnr
-    elif opts.z_var == 'max(newsnr_sgveto)':
-        z = trigs.newsnr_sgveto
-    elif opts.z_var == 'max(newsnr_sgveto_psdvar)':
-        z = trigs.newsnr_sgveto_psdvar
+    z = trigs.'%s'%opts.z_var
```
Comment (ahnitz) [inline bin/hdfcoinc/pycbc_plot_singles_vs_params]: z = getattr(trigs, opts.z_var)

**Comment by ahnitz** [comment]
@sum33it Has tested he can generate by hand plots from this patch. 

This patch requires changes to config files which will call the updated version of this code.

the sngl-stat option now uses the same name as the dict in the pycbc/events/stat module instead of things like "max(snr)" you give just "snr". 

The config file in https://github.com/gwastro/pycbc-config/blob/master/test/plotting.ini should be updated in a separate PR against the pycbc-config repo.

**Comment by tdent** [comment]
@ahnitz @sum33it I was completely unaware of changes - which silently break most of the plots unless the input option for z-var is changed - and none of the O2 or O3 plotting ini files seem to have been updated to include this change.  The change is also not evident in the help message. 

How were we supposed to know that we should change the ini files for ongoing analysis and development?

**Comment by tdent** [comment]
Also, there is no fail-through for the z-var value:
`if opts.z_var == 'density':
    norm = LogNorm()
    hb = ax.hexbin(x, y, norm=norm, vmin=1, **hexbin_style)
    fig.colorbar(hb, ticks=LogLocator(subs=range(10)))
elif opts.z_var in sngl_statistic_dict:
`

if z_var is *not* in sngl_statistic_dict, it just does nothing and produces a mysterious empty plot - whereas it should report an error.

**Comment by ahnitz** [comment]
@tdent You should track changes to the ini files. They have been updated here 
https://github.com/gwastro/pycbc-config/tree/master/test
This is the location which should always work with master.

**Pull request merged**